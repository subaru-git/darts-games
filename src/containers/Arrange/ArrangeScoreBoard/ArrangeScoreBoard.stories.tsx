import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import ArrangeScoreBoard from './ArrangeScoreBoard';

export default {
  title: 'Components/ArrangeScoreBoard',
  component: ArrangeScoreBoard,
} as ComponentMeta<typeof ArrangeScoreBoard>;

const Template: ComponentStory<typeof ArrangeScoreBoard> = (args) => <ArrangeScoreBoard {...args} />;

export const Default = Template.bind({});
Default.args = {};
