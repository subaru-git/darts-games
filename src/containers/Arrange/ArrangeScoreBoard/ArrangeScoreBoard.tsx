import React, { FC } from 'react';

type ArrangeScoreBoardProps = {
  score: string[][];
};

const ArrangeScoreBoard: FC<ArrangeScoreBoardProps> = ({ score }) => {
  return (
    <div>
      <table className='w-full text-center'>
        <thead className='bg-gray-50 uppercase'>
          <tr>
            <th scope='col' className='px-3 py-1 md:px-6 md:py-3'>
              Target
            </th>
            <th scope='col' className='px-3 py-1 md:px-6 md:py-3' colSpan={4}>
              Hits
            </th>
          </tr>
        </thead>
        <tbody>
          {score.map((items, i) => (
            <tr key={i} className='border-b'>
              {items.map((item, i) => (
                <td
                  key={i}
                  className={`px-3 py-1 italic md:px-6 md:py-4 ${i === 0 ? 'font-bold' : ''} ${
                    item === 'FINISH'
                      ? 'text-red-700'
                      : item === 'BUST'
                      ? 'text-blue-700'
                      : 'text-gray-500'
                  }`}
                >
                  {item}
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default ArrangeScoreBoard;
