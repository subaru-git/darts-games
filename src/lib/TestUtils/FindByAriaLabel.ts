export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
