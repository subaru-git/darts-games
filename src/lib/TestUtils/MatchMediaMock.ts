import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
