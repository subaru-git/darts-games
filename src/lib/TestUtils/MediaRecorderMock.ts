export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
