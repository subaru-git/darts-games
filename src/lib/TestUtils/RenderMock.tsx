import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
