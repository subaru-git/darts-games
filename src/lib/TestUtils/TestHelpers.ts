import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}