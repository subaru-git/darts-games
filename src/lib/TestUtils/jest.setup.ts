import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
