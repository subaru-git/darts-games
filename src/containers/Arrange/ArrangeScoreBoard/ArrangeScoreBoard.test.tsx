import ArrangeScoreBoard from './ArrangeScoreBoard';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(<ArrangeScoreBoard score={[]} />);
  expect(container).toMatchSnapshot();
});
