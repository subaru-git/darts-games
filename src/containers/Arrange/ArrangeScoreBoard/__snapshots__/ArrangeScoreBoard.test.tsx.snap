// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div>
    <table
      class="w-full text-center"
    >
      <thead
        class="bg-gray-50 uppercase"
      >
        <tr>
          <th
            class="px-3 py-1 md:px-6 md:py-3"
            scope="col"
          >
            Target
          </th>
          <th
            class="px-3 py-1 md:px-6 md:py-3"
            colspan="4"
            scope="col"
          >
            Hits
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
